lib/honeybadger/check_in.rb

```diff
@@ -0,0 +1,87 @@
+module Honeybadger
+  class InvalidCheckinConfig < StandardError; end
+  class CheckInSyncError < StandardError; end
+  
+  class CheckIn
```

Comment:
This seems like a lot of code for what is essentially shuffling hashes back and forth. What about something like this?

```ruby
# frozen_string_literal: true

class CheckIn
  ATTRIBUTES = %w[name slug schedule_type report_period grace_period cron_schedule cron_timezone]

  attr_reader :project_id, :data

  def self.from_config(atts)
    new(atts["project_id"], atts["id"], atts)
  end

  def self.from_remote(project_id, atts)
    new(project_id, atts["id"], atts)
  end

  def initialize(project_id, id: nil, atts: nil)
    @project_id = project_id
    @id = id
    @data = atts.transform_keys(&:to_s).slice(*ATTRIBUTES) if atts.is_a?(Hash)
  end

  def ==(other)
    data == other.data
  end

  def to_json
    data.to_json
  end

  ATTRIBUTES.each do |meth|
    define_method meth do
      data[meth]
    end
  end

  ...
end
```

Reply:
@stympy This was a conscious decision on my part as I usually value the explicitness over brevity but I'll happily go with your approach if that's more of your liking.

Reply:
Got it! Yeah, I think we'd prefer brevity. Thanks for your flexibility!